qmf/src/plugins/messageservices/imap/imapservice.cpp

```diff
@@ -1795,11 +1853,22 @@ void ImapService::updateStatus(const QString &text)
 }
 
 #ifdef USE_KEEPALIVE
+void ImapService::onUpdateLastSyncTime()
+{
+    // start timer again if still in idle mode
+    if (_idling) {
+        _backgroundActivity->wait();
+    } else if (_backgroundActivity->isRunning()){
+        qMailLog(Messaging) << Q_FUNC_INFO << "Stopping keepalive";
+        _backgroundActivity->stop();
```

Comment:
Because subsequent run might be already going when we receive this signal. I.e additional timer goes one for nothing.